lib/selinux/selinux_linux.go

```diff
@@ -131,6 +179,32 @@ func CheckConfiguration(ensureEnforced bool, logger *slog.Logger) error {
 	return nil
 }
 
+// this function attempts to diagnose why Teleport is not running under the correct SELinux domain.
+func diagnoseDomain(procCtx *selinuxContext, logger *slog.Logger) error {
+	if procCtx.user != seSystemUser || procCtx.role != seSystemRole {
+		logger.WarnContext(context.TODO(), "Teleport is not running as the system_u:system_r SELinux user and role, running Teleport as a Systemd service is recommended when --enable-selinux is passed", "selinux_context", procCtx.String())
```

Comment:
Or pass context here.
```suggestion
		logger.WarnContext(context.Background(), "Teleport is not running as the system_u:system_r SELinux user and role, running Teleport as a Systemd service is recommended when --enable-selinux is passed", "selinux_context", procCtx.String())
```